### PR TITLE
Medical GUI - Add keybind to show player's patient information display

### PR DESCRIPTION
**Pull request by amsteadrayle**
**When merged this pull request will:**
- Add a keybind to temporarily show your patient information display

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by amsteadrayle** [comment]
The idea with this is to allow the player to check their medical status without interrupting gameplay or losing control of their character.

I set the default keybind to Ctrl+H. It doesn't conflict with any other defaults that I'm aware of, but it's not particularly ergonomic for using as a quick peek while doing other things. So any suggestions are welcome.

**Comment by amsteadrayle** [comment]
Here's a quick video showing it in use:
https://www.youtube.com/watch?v=MFcGYJ3Z3BQ

**Comment by PabstMirror** [comment]
looks neat
we could maybe use `enableMedicalMenu` setting for this as well?

**Comment by amsteadrayle** [comment]
Oh, true. Although the preview through the interact menu doesn't seem to be affected by that setting. 

Should `canOpenMenu` be added or replace the condition?
```sqf
if !([ACE_player, ACE_player] call FUNC(canOpenMenu)) exitWith {false};
```

**Comment by veteran29** [comment]
It would be nice to have the fade-out time configurable and not have to hold the key all the time to view it. Personally, I would like to press the keybind and have 1-2s to quickly glance over medical state.

**Comment by amsteadrayle** [comment]
I've added a setting to delay the fadeout in a range of 0-5 seconds, defaulting to 1 second.

File: addons/medical_gui/XEH_postInit.sqf [hunk 1]
```diff
@@ -71,6 +71,18 @@ GVAR(selfInteractionActions) = [];
...
+["ACE3 Common", QGVAR(peekMedicalInfoKey), localize LSTRING(PeekMedicalInfo),
+{
+    // Conditions: canInteract
+    if !([ACE_player, objNull, []] call EFUNC(common,canInteractWith)) exitWith {false};
+
+    // Statement
+    [ACE_player, 0] call FUNC(displayPatientInformation);
+    false
+}, {
+    [{QGVAR(RscPatientInfo) cutFadeOut 0.3; false}, [], round GVAR(peekMedicalInfoReleaseDelay)] call CBA_fnc_waitAndExecute;
```
Comment (Mike-MF) [inline addons/medical_gui/XEH_postInit.sqf:83]:
```suggestion
    [{QGVAR(RscPatientInfo) cutFadeOut 0.3; false}, [], GVAR(peekMedicalInfoReleaseDelay)] call CBA_fnc_waitAndExecute;
```

File: addons/medical_gui/initSettings.sqf [hunk 2]
```diff
@@ -126,3 +126,12 @@ private _categoryColors = [ELSTRING(medical,Category), format ["| %1 |", LELSTRI
     true,
     true // isGlobal
 ] call CBA_fnc_addSetting;
+
+[
+    QGVAR(peekMedicalInfoReleaseDelay),
+    "SLIDER",
+    [LSTRING(PeekMedicalInfoReleaseDelay_DisplayName), LSTRING(PeekMedicalInfoReleaseDelay_Description)],
+    [ELSTRING(medical,Category), LSTRING(SubCategory)],
+    [0, 5, 1, 0],
```
Comment (Mike-MF) [inline addons/medical_gui/initSettings.sqf:135]:
```suggestion
    [0, 5, 1, -1],
```

**Review by Mike-MF — COMMENTED** [review]
Alternative way of rounding off the number.

File: addons/medical_gui/initSettings.sqf (hunk 2 above)
Comment (amsteadrayle) [reply addons/medical_gui/initSettings.sqf]: Oh nice! I though there had to be a way to do an integer slider.

**Review by Mike-MF — APPROVED** [review]
Tested, working.

File: addons/medical_gui/stringtable.xml [hunk 3]
```diff
@@ -282,6 +282,15 @@
             <Chinese>開起醫療選單</Chinese>
             <Turkish>Medikal Menüyü Aç</Turkish>
         </Key>
+        <Key ID="STR_ACE_Medical_GUI_PeekMedicalInfo">
+            <English>Peek Medical Info</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_GUI_PeekMedicalInfoReleaseDelay_DisplayName">
+            <English>Medical Info Peek Fadeout Delay</English>
```
Comment (LinkIsGrim) [inline addons/medical_gui/stringtable.xml:289]: Pretty long for a setting name. Maybe `Medical Peek Time`?

File: addons/medical_gui/initSettings.sqf [hunk 4]
```diff
@@ -126,3 +126,12 @@ private _categoryColors = [ELSTRING(medical,Category), format ["| %1 |", LELSTRI
     true,
     true // isGlobal
 ] call CBA_fnc_addSetting;
+
+[
+    QGVAR(peekMedicalInfoReleaseDelay),
+    "SLIDER",
+    [LSTRING(PeekMedicalInfoReleaseDelay_DisplayName), LSTRING(PeekMedicalInfoReleaseDelay_Description)],
+    [ELSTRING(medical,Category), LSTRING(SubCategory)],
+    [0, 5, 1, -1],
```
Comment (LinkIsGrim) [inline addons/medical_gui/initSettings.sqf:135]:
```suggestion
    "TIME",
    [LSTRING(PeekMedicalInfoReleaseDelay_DisplayName), LSTRING(PeekMedicalInfoReleaseDelay_Description)],
    [ELSTRING(medical,Category), LSTRING(SubCategory)],
    [0, 5, 1],
```

**Comment by amsteadrayle** [comment]
Last push fixes this incorrect behavior, where the timeout from an earlier press would override a more recent one.
https://github.com/acemod/ACE3/assets/2516219/5c740a23-029e-41ad-a760-2e5a1346ca92

File: addons/medical_gui/XEH_postInit.sqf [hunk 5]
```diff
@@ -71,6 +73,25 @@ GVAR(selfInteractionActions) = [];
...
+    [ACE_player, 0] call FUNC(displayPatientInformation);
+    false
+}, {
+    GVAR(peekLastOpenedOn) = CBA_missionTime;
+    [{
+        CBA_missionTime - GVAR(peekLastOpenedOn) > GVAR(peekMedicalInfoReleaseDelay)
+    }, {
+        QGVAR(RscPatientInfo) cutFadeOut 0.3;
+        false
+    }] call CBA_fnc_waitUntilAndExecute;
```
Comment (LinkIsGrim) [inline addons/medical_gui/XEH_postInit.sqf:91]:
```suggestion
    if (CBA_missionTime - GVAR(peekLastOpenedOn) > GVAR(peekMedicalInfoReleaseDelay) then {
        [{
            CBA_missionTime - GVAR(peekLastOpenedOn) > GVAR(peekMedicalInfoReleaseDelay)
        }, {QGVAR(RscPatientInfo) cutFadeOut 0.3}] call CBA_fnc_waitUntilAndExecute;
    };
    GVAR(peekLastOpenedOn) = CBA_missionTime;
```

This should prevent creating multiple WUAE's, though it's not that big of a deal.

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**